File: backend/tiggle/src/main/java/com/side/tiggle/domain/member/MemberDto.java
package com.side.tiggle.domain.member;

import com.side.tiggle.domain.member.model.Member;
import org.springframework.security.oauth2.core.user.OAuth2User;

import java.time.LocalDate;

public class MemberDto {
    long id;
    String email;
    String profileUrl;
    String nickname;
    LocalDate birth;

    public static MemberDto fromEntity(Member member) {
        MemberDto dto = new MemberDto();
        dto.id = member.getId();
        dto.birth = member.getBirth();
        dto.email = member.getEmail();
        dto.nickname = member.getNickname();
        dto.profileUrl = member.getProfileUrl();
        return dto;
    }
}

File: backend/tiggle-root/tiggle/src/main/kotlin/com/side/tiggle/domain/member/dto/service/MemberDto.kt
package com.side.tiggle.domain.member.dto.service

import com.side.tiggle.domain.member.dto.controller.MemberResponseDto
import com.side.tiggle.domain.member.model.Member
import java.time.LocalDate

data class MemberDto(
    val id: Long?,
    val email: String,
    val profileUrl: String?,
    val nickname: String,
    val birth: LocalDate?
) {
    companion object {
        @JvmStatic
        fun fromEntityToMemberResponseDto(member: Member): MemberResponseDto {
            return MemberResponseDto(member.id, member.email, member.profileUrl, member.nickname, member.birth)
        }
    }
}